pandas/io/parsers.py

```diff
@@ -3421,11 +3433,14 @@ def get_rows(self, n, skiprows=None):
         self.buffer = iter(buffer_rows)
         return detect_rows
 
-    def detect_colspecs(self, n=100, skiprows=None):
+    def detect_colspecs(self, n=None, skiprows=None, infer_nrows=100):
+        # infer_nrows replaces n, see GH15138
         # Regex escape the delimiters
         delimiters = ''.join(r'\%s' % x for x in self.delimiter)
         pattern = re.compile('([^%s]+)' % delimiters)
-        rows = self.get_rows(n, skiprows)
+        if n:
```

Comment:
Since this is an internal method only and not part of the api I wonder if it wouldn't be better to just replace the `n` keyword with `infer_rows`

Reply:
That sounds reasonable to me, @WillAyd. Would it be better to leave the keyword order as-is with `n` removed or should I reorder like this `def detect_colspecs(self, infer_nrows=100, skiprows=None):`? Or does that really not matter at all?

Reply:
I'd go with the latter unless @jreback has a differing opinion

Reply:
i think ok to reorder that is an internal method